mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -66,6 +68,23 @@ def message_post(
                         subject = rendered_subject_template + subject
                     elif template.position == "append_after":
                         subject += rendered_subject_template
+                    elif template.position == "inside_replace":
+                        try:
+                            rendered_subject_to_replace = self.env[
+                                "mail.template"
+                            ]._render_template(
+                                template_src=template.subject_to_replace,
+                                model=self._name,
+                                res_ids=[self.id],
+                            )[self.id]
+                            if rendered_subject_to_replace:
+                                # To avoid empty string replacements
+                                subject = subject.replace(
+                                    rendered_subject_to_replace,
+                                    rendered_subject_template,
+                                )
+                        except Exception:
```

Comment:
issue: this is a very broad `except` clause, that could catch any kind of errors, and passes unnoticed. That's most likely a bad idea. If the user has a buggy template, there'll be no way to know why it's failing.

It should either raise something that the user can react to, or at least log something so a sysadmin can inspect the reason of possible failures.

Reply:
I totally agree, although the same “_solution_” already exists at https://github.com/OCA/social/blob/a642059ed937fd94b1339a388c0d4f82a4ae9292/mail_notification_custom_subject/models/mail_thread.py#L70.

I'm going to remove the try/except directly so that if there is an error it will happen and it will be known.

Reply:
I see...

Well, controlling the exception makes kind of sense, because if batch-mailing, you don't want the whole queue to fail just because a single mail failed.

At least I'd log the error.

Reply:
Ok, changed (again)